confmap/provider.go

```diff
@@ -98,6 +98,8 @@ type ChangeEvent struct {
 	// Error is nil if the config is changed and needs to be re-fetched.
 	// Any non-nil error indicates that there was a problem with watching the config changes.
 	Error error
+	// prevent unkeyed literal initialization
+	_ struct{}
```

Comment:
is this a breaking change?

Reply:
I don't think so:

> OpenTelemetry authors reserve the right to introduce API changes breaking compatibility between minor versions in the following scenarios:
> 
> Struct literals. It may be necessary to add new fields to exported structs in the API. Code that uses unkeyed struct literals (such as pkg.T{3, "x"}) to create values of these types would fail to compile after such a change. However, code that uses keyed literals (pkg.T{A: 3, B: "x"}) will continue to compile. We therefore recommend using OpenTelemetry collector structs with the keyed literals only.

from https://github.com/open-telemetry/opentelemetry-collector/blob/main/VERSIONING.md#general-go-api-considerations